pymeasure/instruments/nf/li5660.py

```diff
@@ -0,0 +1,1665 @@
+        """ A property that allow user to set the current-voltage conversion gain for current input.
+            1: IE6 Conversion gain 1 MV/A, 1uAmax
+            100: IE8 Conversion gain 100 MV/A, 10nAmax """
+        self.input_gain = g
+        log.info("The current-voltage conversion gain is set to {}".format(self.input_gain))
+
+    def impedance(self, z=50):
+        """ A property that allow user to set HF terminal input impedance.
+            Rounding is applied to arbitrary values specified. """
+        self.input_impedance = z
+        log.info("The HF terminal input impedance is set to {}".format(self.input_impedance))
+
+    def grounding(self, gud="Ground"):
+        """ A property that allow user to set grounding of the signal input connectedor's outer conductor. """
+        self.input_low = gud
+        log.info("The grounding of the signal input connector's outer conductor is set to {}".format(self.input_low))
+
+    def input_offset(self, offset="Once"):
+        """ A property that allow user to set the PSD input offset. 
+            On: Enables continuius automatic adjustment of PSD input offset.
+            Off: Disable continuous automatic adjustment of PSD input offset.
+            Once: PSD input offset is autumatically adjusted just once.
+            Resotre: Disables PSD input offset adjustment and restores the factory default setting. """
+        if offset=="Once":
+            self.write(":INP:OFFS:AUTO:ONCE")
+        elif offset == "Restore":
+            self.write(":INP:OFFS:RST")
+        else:
+            self.input_offset_auto = offset
+        log.info("The input PSD offset is set to {}".format(self.input_offset_auto))
+
+    def response(self, time=200E-3):
+        """ A property that allow user to set response time for the PSD input offset continuous auto adjustment function.
+            Rounding is applied to arbitrary values specified. """
+        self.input_offset_stime = time
+        log.info("The response time is set to {}".format(self.input_offset_stime))
+
+    def signal_type(self, reference="Sine"):
+        """ A property that allow user to set the reference signal wavefrom.
+            Sine: Sinusoid wave.
+            TTL Rising: TTL level rising edge.
+            TTL Falling: TTL level falling edge. """
+        self.input_type = reference
+        log.info("The input signal type is set to {}".format(self.input_type))
+
+    def delete_memory(self, num=None):
+        """ A property that allow user to clears the contents of the specified configuration memory. """
+        if num is None:
+            return "Please specified whick memory you want to clear."
+        else:
+            self.write(":MEM:STAT:DEL {}".format(num))
+        log.info("{} memory has been cleared.")
+
+    def name_memory(self, name:str, num:int):
+        """ A property that allow uset to changes the name of the specified configuration memory. """
+        if name is None or num is None:
```

Comment:
I think this is not required, since both `name` and `num` do not have `None` as default value.

Reply:
Also, we [generally](https://pymeasure.readthedocs.io/en/latest/dev/coding_standards.html#python-style-guides) don't use type hints in the signature - please put them in the docstring, instead.